### PR TITLE
[DI] Fix accepting null as default env param value

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |  #20447
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by stof** [comment]
but shouldn't we allow it ? It would be useful in case some config treats null as being "don't enable this feature", which is the case for the database_url in Doctrine (allowing to configure with username/password/host locally when not setting the URL)

**Comment by nicolas-grekas** [comment]
Could be, but should be allowed for all parameters then (for consistency, and because we use the same code paths for all params, env or not)

**Comment by stof** [comment]
normal parameters (non-env-based) allow to store ``null`` (they also allow to store arrays, contrary to env parameters)

**Comment by nicolas-grekas** [comment]
Indeed, I mangled here :) Fixed!

File: src/Symfony/Component/DependencyInjection/Tests/ParameterBag/EnvPlaceholderParameterBagTest.php [hunk 1]
```diff
@@ -119,6 +119,15 @@ public function testResolveEnvCastsIntToString()
         $this->assertSame('2', $bag->all()['env(int)']);
     }
 
+    public function testResolveEnvAllowsNull()
+    {
+        $bag = new EnvPlaceholderParameterBag();
+        $bag->get('env(NULL)');
```
Comment (javiereguiluz) [inline src/Symfony/Component/DependencyInjection/Tests/ParameterBag/EnvPlaceholderParameterBagTest.php]: It's a bit confusing in this test to use `NULL` as the var name. What about replacing it by `NULL_VAR` ? (and `Null_Var` in the example below)

**Comment by dunglas** [comment]
👍

File: src/Symfony/Component/DependencyInjection/Tests/ParameterBag/EnvPlaceholderParameterBagTest.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Tests/ParameterBag/EnvPlaceholderParameterBagTest.php]: updated

**Comment by xabbuh** [comment]
👍 

Status: Reviewed

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**